chrome correction

ElFaboon59 opened the pull request:
The following error is raised when a specific size is choosen in a Cerberus robot. The execution report will fail with this message. Several tests has been done, and the error is triggered systematically :

**The test case failed to be executed. Could not start Robot Server. org.openqa.selenium.WebDriverException: unknown error: failed to change window state to normal, current state is maximized**

Configuration :

- Cerberus on glassfish 4 ( 3.10, 3.11, 3.12-snapshot )
- Selenium , selenium/standalone-chrome:3.141.59-mercury image on docker ( version 18.09.4, build d14af54266 )

It comes from the fact that the chrome node is started maximized and thus , cannot be set at the desired dimension. I tried several configuration of Selenium node and robot without success.

The provided fix sets the dimension and maximize command as chrome options, if the screen size is defined in robot configuration panel. It also disable the setcreensize() call that comes afterwards, specifically for the chrome browser, as the size is passed in node start option.

Feel free to contact me if needed.

nicodeur commented on source/src/main/java/org/cerberus/engine/execution/impl/SeleniumServerService.java:
```diff
@@ -332,11 +332,14 @@ public void startServer(TestCaseExecution tCExecution) throws CerberusException
-                    setScreenSize(driver, screenWidth, screenLength);
-                    LOG.debug("Selenium resolution Activated : " + screenWidth + "*" + screenLength);
+                
+                if(!tCExecution.getBrowser().equalsIgnoreCase("chrome")) {
```
is it possible to use the constant BrowserType.CHROME ?

nicodeur commented on source/src/main/java/org/cerberus/engine/execution/impl/SeleniumServerService.java:
```diff
@@ -584,7 +587,19 @@ private DesiredCapabilities setCapabilityBrowser(DesiredCapabilities capabilitie
                  */
                 ChromeOptions options = new ChromeOptions();
                 // Maximize windows for chrome browser
+               
```
Is it possible to avoid duplication code, and just precise `--window-size` and `--headless` only for chrome ?

It will allow to delete condition for chrome on previous changement.

ElFaboon59 commented:
Hello Nicolas,

Thanks for your answer and remarks. Concerning the browser type constant, i
have made the change, no problem.

Regarding your second remark, the code is already contained in a "if"
specifically for chrome browser.

If you have any further remarks , please let me know before i submit a new
submit request.

Regards?

Le mar. 16 avr. 2019 à 12:00, Nicolas Deblock <notifications@github.com> a
écrit :

> *@nicodeur* requested changes on this pull request.
>
> Hi,
>
> Thanks for your pull request !
>
> I have some propositions to increase your code. What do you think about it
> ?
>
> Regards.
>
> Nicolas
> ------------------------------
>
> In
> source/src/main/java/org/cerberus/engine/execution/impl/SeleniumServerService.java
> <https://github.com/cerberustesting/cerberus-source/pull/1964#discussion_r275759487>
> :
>
> > @@ -332,11 +332,14 @@ public void startServer(TestCaseExecution tCExecution) throws CerberusException
>                   */
>                  String targetScreensize = getScreenSizeToUse(tCExecution.getTestCaseObj().getScreenSize(), tCExecution.getScreenSize());
>                  LOG.debug("Selenium resolution : " + targetScreensize);
> -                if ((!StringUtil.isNullOrEmpty(targetScreensize)) && targetScreensize.contains("*")) {
> -                    Integer screenWidth = Integer.valueOf(targetScreensize.split("\\*")[0]);
> -                    Integer screenLength = Integer.valueOf(targetScreensize.split("\\*")[1]);
> -                    setScreenSize(driver, screenWidth, screenLength);
> -                    LOG.debug("Selenium resolution Activated : " + screenWidth + "*" + screenLength);
> +
> +                if(!tCExecution.getBrowser().equalsIgnoreCase("chrome")) {
>
> is it possible to use the constant BrowserType.CHROME ?
> ------------------------------
>
> In
> source/src/main/java/org/cerberus/engine/execution/impl/SeleniumServerService.java
> <https://github.com/cerberustesting/cerberus-source/pull/1964#discussion_r275760376>
> :
>
> > @@ -584,7 +587,19 @@ private DesiredCapabilities setCapabilityBrowser(DesiredCapabilities capabilitie
>                   */
>                  ChromeOptions options = new ChromeOptions();
>                  // Maximize windows for chrome browser
> +
>
> Is it possible to avoid duplication code, and just precise --window-size
> and --headless only for chrome ?
>
> It will allow to delete condition for chrome on previous changement.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/cerberustesting/cerberus-source/pull/1964#pullrequestreview-227145662>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AL9yKB0yjBjK0bbmSgoKWW4SzUq39gNbks5vhbtGgaJpZM4csRV_>
> .
>

ElFaboon59 commented:
Hello,

Change has been commited.

Regards
